altlas/Assets/scripts/MapGenerator.cs

```diff
@@ -290,11 +291,65 @@ public void spawnStacksInRow(float horLength, float verLength, List<List<MapData
         }
     }
 
-    /**
-     * spawnStackInRow with fixed drawer length
-     * */
-    public void spawnRowInDrawer(MapData[][] stacks, string category) {
-        spawnStacksInRow(DRAWER_HOR_LEN, DRAWER_VER_LEN, stacks, category);
+    void setLabelOnDraw(KeyValuePair<string, Dictionary<string, List<MapData>>> category)
+    {     
+        GameObject go = new GameObject("myText");
+        go.AddComponent<TextMesh>();
```

Comment:
you could just store the return value of this function into tm instead of calling getComponent